matchms/filtering/derive_formula_from_name.py

```diff
@@ -0,0 +1,51 @@
+import re
+from ..typing import SpectrumType
+
+
+def derive_formula_from_name(spectrum_in: SpectrumType, remove_formula_from_name=True) -> SpectrumType:
+    """Detect and remove misplaced formula in compound name and add to metadata.
+
+    Method to find misplaced formulas in compound name based on regular expression.
+    Args:
+    ----
+    spectrum_in: SpectrumType
+        Input spectrum.
+    remove_formula_from_name: bool
+        Remove found formula from compound name if set to True. Default is True.
+    """
+    if spectrum_in is None:
+        return None
+
+    spectrum = spectrum_in.clone()
+
+    # Get compound name
```

Comment:
Confusing comment. Removing it increases readibility to me. And code duplication.

Reply:
Removed the comment.